tests/test_mock_controller.py

```diff
@@ -10,35 +11,111 @@ class MockTestCase(asynctest.TestCase):
         self.writer = None
         self.port = 5000
         self.mock_ctrl = None
+        self.data = None
 
         self.mock_ctrl = Dome.MockDomeController(port=self.port)
         asyncio.create_task(self.mock_ctrl.start())
         await asyncio.sleep(1)
 
         rw_coro = asyncio.open_connection(host="127.0.0.1", port=self.port)
-        self.reader, self.writer = await asyncio.wait_for(rw_coro, timeout=5)
+        self.reader, self.writer = await asyncio.wait_for(rw_coro, timeout=1)
+        self.data = await self.read()
+        self.assertComponent("AMCS", {"status": "Stopped", "positionActual": 0})
+        self.assertTBD("ApCS")
+        self.assertTBD("LCS")
+        self.assertTBD("LWCS")
+        self.assertTBD("ThCS")
+        self.assertTBD("MonCS")
+
+    async def read(self):
+        """ Utility function to read a string from the reader and unmarshal it
+        """
         read_bytes = await asyncio.wait_for(
-            self.reader.readuntil("\n".encode()), timeout=5
+            self.reader.readuntil("\r\n".encode()), timeout=1
         )
-        read_str = read_bytes.decode()
-        self.assertTrue("AMCS:Stopped" in read_str)
-        self.assertFalse("MonCS:Stopped" in read_str)
+        data = yaml.safe_load(read_bytes.decode())
+        return data
+
+    async def write(self, st):
+        """ Utility function to write a string to the writer
+        """
+        self.writer.write(f"{st}\r\n".encode())
+        await self.writer.drain()
+
+    def assertComponent(self, component, expected):
+        """ Asserts that the values of the AMCS parameter data are as expected
```

Comment:
It seems a shame that this can't be used to assert OK replies. I mean it *can* but it's a weird thing to do. I wonder if you could refactor a bit to have an assertComponent and assertOKReply (icky name) that both call the same underlying code to check the main data.

Reply:
AMCS -> Thanks, I forgot to refactor the comment.
Thanks for being picky. I got rid of the space.

I'll look into the possibility of checking OK with the same code. My idea was to write a generic function to check the status of all 6 sub-components and never considered checking OK (and ERROR for that matter) with it as well.